docs/deploying.md

```diff
@@ -0,0 +1,100 @@
+# Deploying
+
+When time comes to share your project, you have many options for deploying it for others to see. Framework is compatible with many static site hosts and automation environments. In this guide we’ll focus on deploying to Observable with GitHub Actions.
+
+## Manually deploying to Observable
+
+If you don’t already have a project to deploy, you can create one by following [getting-started](./getting-started). First, make sure that your project builds without error:
+
+```sh
+$ npm run build
+```
+
+Once that is done you can deploy to Observable with the command
+
+```sh
+$ npm run deploy
+```
+
+The first time you run this, it will prompt you for details needed to set up the project on the server, such as the project's _slug_ (which determines its URL), and the access level. If you don’t already have an Observable account or aren’t signed in, this command will also guide you through setting that up.
+
+When the deploy command finishes, it prints a link to observablehq.cloud where you can view your deployed project. If you choose “private” as the access level, you can now share that link with anyone who is a member of  your workspace. If you chose “public”, you can share that link with anyone and they’ll be able to see your Framework project.
+
+<div class="note">The deploy command will create a file at <code>docs/.observablehq/deploy.json</code> which you should commit to git. If you have configured a source root besides `docs/`, the file will be placed there instead. This file is used to store the project to deploy to, and is required for automated deploys.</div>
+
+## Automated deploys to Observable
+
+To set up automatic deploys, we’ll be using [GitHub actions](https://github.com/features/actions). In your git repository, create and commit a file at `.github/workflows/deploy.yml`. Here is a starting example:
+
+```yaml
+name: Deploy
+on:
+  # Run this workflow whenever a new commit is pushed to main.
+  push: {branches: [main]}
+  # Run this workflow once per day, at 10:15 UTC
+  schedule: [{cron: "15 10 * * *"}]
+  # Run this workflow when triggered manually in GitHub's UI.
+  workflow_dispatch: {}
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
```

Comment:
the current version of https://github.com/actions/setup-node is `@v4`; can we recommend it?

Reply:
I was just copying from the BI repo. I think I have every thing using the latest versions now.

Reply:
In my testing, `@v3` works, but it raises an annotation warning, which is resolved if you switch all versions to `@v4` (See [deploy example](https://github.com/cookwcode/restaurant/actions/runs/8275768772))

Reply:
I've updated `actions/setup-node` and `actions/cache` to both use `@v4`.